src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/JsonHubProtocol.java

```diff
@@ -37,7 +37,10 @@ public TransferFormat getTransferFormat() {
 
     @Override
     public HubMessage[] parseMessages(String payload, InvocationBinder binder) {
-        if (payload != null && !payload.substring(payload.length() - 1).equals(RECORD_SEPARATOR)) {
+        if(payload == null || payload.length() == 0 ){
```

Comment:
Is payload ever actually null? It looks like parseMessages would have thrown before this change if it was.